### PR TITLE
Use backticks on some field names.

### PR DESCRIPTION
**Pull request by thomasvl**

### CODE REVIEW

**Comment by thomasvl** [comment]
This will basically fix #531, since it doesn't look like backticks are really doable for enum and message names.

File: Sources/PluginLibrary/NamingUtils.swift [hunk 1]
```diff
@@ -70,6 +69,32 @@ fileprivate let reservedTypeNames: Set<String> = {
...
+ * Many Swift reserved words can be used as fields names if we put
+ * backticks around them:
+ */
+fileprivate let quotableFieldNames: Set<String> = {
+  () -> Set<String> in
+  var names: Set<String> = []
+
+  // We don't need to protect all of these keywords, just the ones
+  // that interfere with enum cases:
+  // names = names.union(swiftKeywordsReservedInParticularContexts)
```
Comment (allevato) [inline Sources/PluginLibrary/NamingUtils.swift]: Can we do something with these commented lines (and below)? It's not clear why they're here.

File: Sources/PluginLibrary/NamingUtils.swift [hunk 2]
```diff
@@ -392,6 +416,10 @@ public enum NamingUtils {
       return s + "_p"
     } else if reservedFieldNames.contains(basedOn) {
       return s + "_p"
+    } else if basedOn == s && quotableFieldNames.contains(basedOn) {
+      // backticks are only used on the base names, if we're sanitizing based on something else
+      // this is skiped (the "hasFoo" doesn't get backticks just because the "foo" does).
```
Comment (allevato) [inline Sources/PluginLibrary/NamingUtils.swift]: s/skiped/skipped/

File: Sources/PluginLibrary/NamingUtils.swift (hunk 1 above)
Comment (thomasvl) [inline Sources/PluginLibrary/NamingUtils.swift]: Done.  Turns out I never cycled back to finish validation, these ones didn't need to be here.

File: Sources/PluginLibrary/NamingUtils.swift (hunk 2 above)
Comment (thomasvl) [inline Sources/PluginLibrary/NamingUtils.swift]: Done.

**Pull request merged**